src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -948,62 +948,25 @@ private bool VerifyNamedArgumentsAfterFixed(Expr args)
 
         ////////////////////////////////////////////////////////////////////////////////
         // Report a bad operator types error to the user.
-        private ExprOperator BadOperatorTypesError(ExpressionKind ek, Expr pOperand1, Expr pOperand2)
+        private RuntimeBinderException BadOperatorTypesError(Expr pOperand1, Expr pOperand2)
```

Comment:
>BadOperatorTypesError [](start = 39, length = 21)

nit: Why returning the exception and then throwing it in the caller in all cases? why not throwing it here?

Reply:
It lets me get rid of the case in https://github.com/dotnet/corefx/commit/bfb67c0152f6a5e47c0f7b5d617c70552fe81567#diff-5abcb553584977a02323beccc54f72c2L432 where a bad expression would (if it were reachable) be returned, and without doing something like `Debug.Fail("How are we even here?"); throw Error.InternalCompilerError();` and makes the other calls more obviously points that throw.

That's not a big gain here, but it's in keeping with a general approach introduced in #22748 of treating methods that always throw as exception factories and throwing at the callsite, making the point after it more obviously unreachable which in some cases is a bigger gain.